src/ghc.jl

```diff
@@ -0,0 +1,98 @@
+"""
+    GHC(k, variogram)
+
+Geostatistical Hierarchical Clustering.
+
+Long description....
+
+## References
+...
+"""
+struct GHC <: PartitionMethod
+  variogram::V
+  k::Int
+end
```

Comment:
Notice that I wrote this snippet over Zulip as an example. I need to review the paper again to double check the exact parameters of the method.

Reply:
Ok, yes additionally there are some different choices (like type of kernel, etc) which I assume we will make kwargs with a default